src/helpers/script_helpers.sh

```diff
@@ -117,6 +145,15 @@ function run {
     preconditions "$@"
     steps > "$STEPS_FILE"
     run_steps "$STEPS_FILE" undoable
+
+    # Apply global previous-branch restoration
+    # only if we're at the git root.
+    # Any branch restoration logic for the other
+    # case needs to be handled in a custom way by
+    # the command itself
```

Comment:
That was only for `git sync`. And those got added to the steps definitions so by this time we will be back to the initial folder. Looking at this now I believe this actually needs to be added to the steps for each command instead of out here as otherwise we may not get unexpected behavior when there are conflicts.